octoprint_PrintTimeGenius/static/js/PrintTimeGenius.js

```diff
@@ -70,19 +70,25 @@ $(function() {
         }
       }
       return results;
-    }
+    };
 
-    self.FileList = ko.pureComputed(function() {
-      return self.recurseFiles(self.filesViewModel.allItems()).slice()
-          .sort(function(a,b) {
-            if (_.has(a, "gcodeAnalysis.progress") !=
-                _.has(b, "gcodeAnalysis.progress")) {
-              return (_.has(a, "gcodeAnalysis.progress") -
-                      _.has(b, "gcodeAnalysis.progress"));
-            }
-            return a.path.localeCompare(b.path);
-          });
-    });
+    self.theFiles = function(items) {
+    	let results = [];
+    	let queue = [{children: items}];
+
+    	while (queue.length > 0) {
+    		item = queue.shift();
+    		results.push.apply(results, item.children.filter(item => item["type"] == "machinecode"));
```

Comment:
This is odd, I've never used the apply of the push function.  How is it different and better from using push directly?

Reply:
`function.apply` allows you to pass an array instead of individual arguments, that's all

Reply:
You can do that with the spread operator now, right?  https://stackoverflow.com/a/35167699/4454

Reply:
ah true! I'll update this